packages/@aws-cdk/aws-cloudwatch/lib/metric.ts

```diff
@@ -395,6 +397,26 @@ export class Metric implements IMetric {
 
     return list;
   }
+
+  //rules from here: https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-cw-dimension.html
+  private validateDimensions(dims: DimensionHash): void {
+    if (Object.keys(dims)?.length > 10) {
+      throw new Error('The maximum number of dimensions is 10');
+    }
+
+    Object.keys(dims).sort().map(key => {
+      if (dims[key] === undefined || dims[key] === null) {
+        throw new Error(`Dimension value of '${dims[key]}' is invalid`);
```

Comment:
I didn't see in the docs that undefined is not allowed, assuming you verified it in deployment?

Reply:
I ran the code sample in the original issue (#3116), and it failed. I think it is because of the `undefined` value